nova/virt/vmwareapi/vmops.py

```diff
@@ -422,10 +422,17 @@ def _get_extra_specs(self, flavor, image_meta=None):
         if CONF.vmware.reserve_all_memory:
             extra_specs.memory_limits.reservation = int(flavor.memory_mb)
         else:
-            memory_reserved_mb = \
-                utils.get_reserved_memory(flavor)
+            memory_reserved_mb, cpu_reserved = \
+                utils.get_reserved_memory_and_cpu(flavor)
             if memory_reserved_mb > 0:
                 extra_specs.memory_limits.reservation = memory_reserved_mb
+            if cpu_reserved > 0:
+                # NOTE(jakobk): CPU reservations are given in MHz, not in
+                # number-of-CPUs, so we have to translate here.
+                cluster_host_mhz = cluster_util.get_cluster_median_cpu_mhz(
+                                                self._session, self._cluster)
```

Comment:
Do we have to query it on each spawn/resize or could we use some cached value (e.g. from `VCState`)?